datafusion/physical-plan/src/limit.rs

```diff
@@ -51,6 +51,9 @@ pub struct GlobalLimitExec {
     /// Execution metrics
     metrics: ExecutionPlanMetricsSet,
     cache: PlanProperties,
+    /// Whether the limit is order-sensitive
+    /// Such as the child plan is a sort node, then the limit is order-sensitive
+    order_sensitive: bool,
```

Comment:
Just a thought: instead of order sensitive could this store the sort order it requires? If we have the info available and are adding a field that could be more useful in ensuring correctness of the plan is preserved through transformations.

Reply:
Do you mean replace the `order_sensitive` with `required_ordering: Option<LexOrdering>`